prog/pwm/fancontrol

```diff
@@ -390,6 +394,40 @@ function pwmdisable()
 		return 0
 	fi
 
+	# Try to restore pwmN and pwmN_enable value to the same state as before fancontrol start.
+	# Restoring the pwmN value is tried first, before the pwmN_enable mode switch. 
+	# Some chips seem to need this to properly restore fan operation, when activating automatic (2) mode.
+	if [ ${PWM_ENABLE_ORIG_STATE[${1}]} ]
+	then
+		#restore the pwmN value
+		if [ "$DEBUG" != "" ]
+		then
+			echo "Restoring ${1} original value of ${PWM_ORIG_STATE[${1}]}"
+		fi
+		echo ${PWM_ORIG_STATE[${1}]} > ${1} 2> /dev/null
+		# restore the pwmN_enable value, if it is not 1. 
+		# 1 is already set through fancontrol and setting it again might just reset the pwmN value.
+		if [ ${PWM_ENABLE_ORIG_STATE[${1}]} != 1 ]
+		then
+			if [ "$DEBUG" != "" ]
+			then
+				echo "Restoring $ENABLE original value of ${PWM_ENABLE_ORIG_STATE[${1}]}"
+			fi
+			echo ${PWM_ENABLE_ORIG_STATE[${1}]} > $ENABLE 2> /dev/null
+		fi
+		# If pwmN_enable should not be manual (1), check if setting it was successful. 
+		# Checking the pwmN value makes no sense, as it might already have been altered by the chip.
+		if [ ${PWM_ENABLE_ORIG_STATE[${1}]} != 1 ] && [ `cat $ENABLE` = ${PWM_ENABLE_ORIG_STATE[${1}]} ]
```

Comment:
Some of the nesting here is unnecessary - you don't need to check twice for ${PWM_ENABLE_ORIG_STATE[${1}]} != 1